Added setCameraSpec()

LJMU-JackMallo opened the pull request:
Default values declared at the top, method takes json data as argument then checks if res,iso and shutterspeed have been specified, if they have over write the default values, if not continue.

BTWA will need to be altered to send 'iso' and 'shutter_speed' along with 'resolution'.

Apparently you can't directly set iso settings on PiCamera2 so round(100 / 100,1) converts to AnalogueGain 1.0

LJMU-JackMallo commented:
Also the default capture time will be better sent client side imo, so all cameras still capture in sync

ElBeenMachine commented:
@LJMU-JackMallo main.py needs updating, the new util function hasn't been used anywhere

LJMU-JackMallo commented:
sorted whoops

LJMU-JackMallo commented:
cam is now initialised in setCaptureSpec().

Added STILL and STREAM capture modes so we can alter camera settings for different use cases down the line.

Added logging for Resolution , Iso, Shutter speed and camera instances.

Moved all imports to util.py to clean up main.

ElBeenMachine commented:
@LJMU-JackMallo Looks good, I've just added an if statement to each capture method to ensure that if no time is specified, then it default to immediately capturing the image/ending the stream

LJMU-JackMallo commented:
good stuff

ElBeenMachine reviewed (approved):
Passed tests
